### ISSUE TITLE
[Android] Investigate implementing `autoSensitive` sensitive content mode

### ISSUE THREAD

**Issue by camsim99** [flutter/flutter#160879]
The initial implementation of sensitive content support on Android (to address https://github.com/flutter/flutter/issues/150218) will not implement the `autoSensitive` mode, where Android uses a heuristic based on autofill hints to detect sensitive content, see https://developer.android.com/reference/android/view/View#CONTENT_SENSITIVITY_AUTO. Once the initial implementation lands, we should consider implementing this mode, which may require usage of additional APIs.

@gmackall has already done some investigation as to how this may be done in https://github.com/flutter/engine/compare/main...gmackall:engine:content_sensitivity_auto_poc.
